File: credit.py
counter1 = []
counter2 = []

card_number = input("Number: ")

while not card_number.isdigit():

    card_number = input("Number: ")

card_number = str(card_number)

for i in range(len(card_number)):

    if i % 2 != 0:

        counter1.append(int(card_number[i]))

    else:

        counter2.append(int(card_number[i]))

for i in range(len(counter1)):

    if counter1[i] * 2 > 10:

        pull_apart = str(counter1[i] * 2)
        counter1[i] = int(pull_apart[0])
        counter1.append(int(pull_apart[1]))

    else:

        counter1[i] = counter1[i] * 2

total_num = sum(counter1) + sum(counter2)

valid_invalid = False

if total_num % 10 == 0:

    valid_invalid = True

total_num = str(total_num)

# VISA
if card_number == '4' and valid_invalid == True:

    print("VISA\n")

# MASTERCARD
elif card_number[0] == "5" and card_number[1] in ['1', '2', '3', '4', '5'] and valid_invalid == True:
    
    print("MASTERCARD\n")

# AMERICAN EXPRESS
elif card_number[0] == '3' and card_number[1] in ['4', '7'] and valid_invalid == True:

    print("AMEX\n")

else:
    print("INVALID")

File: credit.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{

    // Declaring variables
    int p13;
    int p14;
    int p15;
    int p16;

    int counter1 = 0;
    int counter2 = 0;
    int last_num = 0;
    int last_num_counter2 = 0;

    // User input
    long card_num = get_long("Number: ");

    // Get size of long
    int size = floor(log10(card_num)) + 1;

    // checksum
    for (int i = 0; i < 16; i++)
    {
        last_num = card_num % 10;

        if (i % 2 == 0)
        {
            counter1 += last_num;
        }

        else
        {
            if ((last_num * 2) >= 10)
            {
                last_num_counter2 = (last_num * 2) % 10;
                counter2 += last_num_counter2;
                counter2 += ((last_num * 2) - last_num_counter2) / 10;
            }

            else
            {
                counter2 += last_num * 2;
            }
        }

        card_num = (card_num - (card_num % 10)) / 10;

        // log some numbers
        if (i == 12)
        {
            p13 = last_num;
        }

        else if (i == 13)
        {
            p14 = last_num;
        }

        else if (i == 14)
        {
            p15 = last_num;
        }

        else if (i == 15)
        {
            p16 = last_num;
        }

    }

    int total_num = counter1 + counter2;

    // type of credit card

    // VISA
    if (((p13 == 4 && size == 13) || (p16 == 4 && size == 16)) && total_num % 10 == 0)
    {
        printf("VISA\n");
    }

    // MASTERCARD
    else if ((p16 == 5 && (p15 == 1 || p15 == 2 || p15 == 3 || p15 == 4 || p15 == 5)) && size == 16 && total_num % 10 == 0)
    {
        printf("MASTERCARD\n");
    }

    // AMERICAN EXPRESS
    else if (p15 == 3 && (p14 == 4 || p14 == 7) && size == 15 && total_num % 10 == 0)
    {
        printf("AMEX\n");
    }

    else
    {
        printf("INVALID\n");
    }
}
